python/lsst/pipe/tasks/htmIndexEngine.py

```diff
@@ -0,0 +1,352 @@
+            doc = "Name of Dec column",
+    )
+    mag_column_list = pexConfig.ListField(
+            dtype = str,
+            default = [],
+            doc = """The values in the reference catalog are assumed to be in AB magnitudes.
+List of column names to use for photometric information.  At least one entry is required."""
+    )
+    mag_err_column_map = pexConfig.DictField(
+            keytype = str,
+            itemtype = str,
+            default = {},
+            doc = "A map of magnitude column name (key) to magnitude error column (value)."
+    )
+    is_photometric_name = pexConfig.Field(
+            dtype = str,
+            default = '',
+            doc = 'Name of column stating if satisfactory for photometric calibration (optional).'
+    )
+    is_resolved_name = pexConfig.Field(
+            dtype = str,
+            default = '',
+            doc = 'Name of column stating if the object is resolved (optional).'
+    )
+    is_variable_name = pexConfig.Field(
+            dtype = str,
+            default = '',
+            doc = 'Name of column stating if the object is measured to be variable (optional).'
+    )
+    id_name = pexConfig.Field(
+            dtype = str,
+            default = '',
+            doc = 'Name of column to use as an identifier (optional).'
+    )
+    extra_col_names = pexConfig.ListField(
+            dtype = str,
+            default = [],
+            doc = 'Extra columns to add to the reference catalog.'
+    )
+    header_lines = pexConfig.Field(
+            dtype = int,
+            default = 0,
+            doc = 'Number of lines to skip when reading the text reference file.'
+    )
+    colnames = pexConfig.ListField(
+            dtype = str,
+            default = [],
+            doc = """An ordered list of column names to use in ingesting the catalog.  With an empty
+list, column names will be discovered from the first line after the skipped header lines."""
+    )
+    delimiter = pexConfig.Field(
+            dtype = str,
+            default = ',',
+            doc = 'Delimiter to use when reading text reference files.  Comma is default.'
+    )
+    level = pexConfig.Field(
+            dtype = int,
+            default = 8,
+            doc = 'Default HTM level.  Level 8 gives ~0.08 sq deg per trixel.',
+    )
+
+    def validate(self):
+        pexConfig.Config.validate(self)
+        if not (self.ra_name and self.dec_name and self.mag_column_list):
+            raise ValueError("RA column name, Dec column name and at least one magnitude column must be"+\
+                             " supplied.")
+        if len(self.mag_err_column_map) > 0 and not len(self.mag_column_list) == len(self.mag_err_column_map):
+            raise ValueError("If magnitude errors are provided, all magnitudes must have an error column")
+
+class IngestReferenceRunner(pipeBase.TaskRunner):
+    """!Task runner for the reference catalog ingester
+    """
+    def run(self, parsedCmd):
```

Comment:
This implementation removes all support for `multiprocessing`, which is good because the design doesn't permit parallelism, but you need to tell the user that (and catch any attempt to set `-j`).
This also removes the traceback squashing, which may be less desirable.